blob/store.go

```diff
@@ -24,16 +24,39 @@ import (
 	"sort"
 )
 
-// A KV represents a mutable key-value store in which each value is identified
-// by a unique, opaque string key.  An implementation of KV is permitted (but
-// not required) to report an error from Put when given an empty key.  If the
-// implementation cannot store empty keys, it must report ErrKeyNotFound when
-// operating on an empty key.
+// A Store represents a collection of key-value namespaces ("keyspaces")
+// identified by string labels. Each keyspace in a store is logically distinct;
+// the keys from one space are independent of the keys in another.
+//
+// Implementations of this interface must be safe for concurrent use by
+// multiple goroutines. When constructing a new store, the caller should
+// arrange to call [blob.Close] on it when it is no longer in use, to give the
+// implementation an opportunity to clean up any internal state.
+type Store interface {
+	// Keyspace returns a key space on the store.
+	Keyspace(name string) (KV, error)
+
+	// Sub returns a new Store subordinate to the receiver.  A substore shares
+	// logical storage with its parent store, but keyspaces derived from the
+	// substore are distinct from keyspaces of the parent store or any other
+	// substores derived from it.
+	Sub(name string) Store
+}
+
+// A KV represents a mutable set of key-value pairs in which each value is
+// identified by a unique, opaque string key.  An implementation of KV is
+// permitted (but not required) to report an error from Put when given an empty
+// key.  If the implementation cannot store empty keys, it must report
+// ErrKeyNotFound when operating on an empty key.
 //
 // Implementations of this interface must be safe for concurrent use by
 // multiple goroutines.  Moreover, any sequence of operations on a KV that does
 // not overlap with any Delete executions must be linearizable.[1]
 //
+// When constructing a new KV, the caller should arrange to call [blob.Close]
```

Comment:
oh huh, KVs closeable too? I suppose that's fine, hadn't thought about it.

Reply:
Maybe we can remove this. I left it here because all the existing KVs need it, but maybe after they are updated to become Stores that won't be necessary. 

In which case Close could become more accurately typed, and/or possibly switched to the StoreCloser interface pattern.

Reply:
My intent is to do some of the actual implementation before merging this, so that things are ready for the switch.